### PR TITLE
Server online

### PR DESCRIPTION
**Pull request by goliver5**

### CODE REVIEW

File: Server/Server.cpp [hunk 1]
```diff
@@ -1,102 +1,328 @@
...
+#include "PacketsDataTypes.h"
+#include "Packethandler.h"
+#include "CircularBuffer.h"
+#include "PacketEnum.h"
 
-int main()
+const short MAXNUMBEROFPLAYERS = 1;
+std::mutex mutex;
+
+struct wow
```
Comment (MorranMogge) [inline Server/Server.cpp:19]: wow is not that descriptive of a name, use "dude" instead

File: Server/Server.cpp [hunk 2]
```diff
@@ -1,102 +1,328 @@
...
+		}
+	}
+};
 
-			socket.send(packet);
-			packet.clear();
-			start = std::chrono::system_clock::now();
+//template <typename T>
+//void sendDataAllPlayets(const T& data, serverData& serverData)
+//{
```
Comment (MorranMogge) [inline Server/Server.cpp:121]: Uncomment functions, otherwise think about deleting them

File: Server/Server.cpp [hunk 3]
```diff
@@ -1,102 +1,328 @@
...
+		if (user->tcpSocket.receive(&datapointer, 256, recv) != sf::Socket::Done)
+		{
+			//error
+		}
+		else
+		{
+			mutex.lock();
+			data->circBuffer->addData(datapointer, recv);
+			mutex.unlock();
+			//std::cout << "size_t: " << std::to_string(sizeof(testPosition)) << std::endl;
```
Comment (MorranMogge) [inline Server/Server.cpp:166]: If you are done with debugging then delete the related comments

File: Server/Server.cpp [hunk 4]
```diff
@@ -1,102 +1,328 @@
...
-	else if (connectionType == "c")
+};
+
+int main()
+{
+	std::string identifier;
+	std::string s = "empty";
+	// Group the variables to send into a packet
+	sf::Packet packet;
+	sf::Packet receivedPacket;
```
Comment (MorranMogge) [inline Server/Server.cpp:222]: remove packets since you use structs instead

File: StortSpelProjekt/Client.h [hunk 5]
```diff
@@ -2,36 +2,97 @@
...
+
+	void tempwrite();
+
+	//returns if a player position has been received
+	bool getIfConnected();
+	int getport()const;
+	int getPlayerId()const;
+
+	//returns the players id on the server
+	int initTEMPPLAYERS();
```
Comment (MorranMogge) [inline StortSpelProjekt/Client.h:69]: temp function? maybe change name our replace with new one?

**Review by MorranMogge — APPROVED** [review]
Fix the comments then you can merge, generally good structure, however there were some commented code which should be removed or given an explanation to why it still is there

File: Server/CircularBuffer.h [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+	bool addData(const void*& incomingData, const std::size_t& recvSize);
+
+	//returns an int enum that is the identifier for the packet type
+	int peekPacketId();//peek
+
+	//returns a char* that points to the packet
+	void* getData();
+
+	//advance the read pointer for data that has been read
+	//void advancePointer();
```
[the remark this answers is not in the document]
Comment (goliver5) [reply Server/CircularBuffer.h:37]: removed the old function

File: Server/Server.cpp (hunk 1 above)
Comment (goliver5) [reply Server/Server.cpp:19]: rip dude

File: Server/Server.cpp (hunk 2 above)
Comment (goliver5) [reply Server/Server.cpp]: uncommented and renamed the function

File: Server/Server.cpp (hunk 3 above)
Comment (goliver5) [reply Server/Server.cpp]: cleaned up the commented code used for debugging

**Review by MorranMogge — APPROVED** [review]
Approved

**Pull request merged**